config/config.go

```diff
@@ -1373,6 +1373,30 @@ func buildInput(name string, tbl *ast.Table) (*models.InputConfig, error) {
 			}
 		}
 	}
+	if node, ok := tbl.Fields["precision"]; ok {
```

Comment:
Lets get #7583 in first.  It shouldn't take 12 lines to grab a single value.

Reply:
I feel that #7583 has too many additional error handling changes and has a high chance of introducing new errors.  Since I'd like to replace our TOML loading with the burntsushi library I'd rather not go through two rounds bug squashing.

Reply:
I've brought over a function from this PR to handle parsing time.Duration from the config with less duplication.

Reply:
That works